controller/resource_blackbox_test.go

```diff
@@ -0,0 +1,279 @@
+}
+
+func (rest *TestResourceREST) SecuredController(identity account.Identity) (*goa.Service, *ResourceController) {
+	svc := testsupport.ServiceAsUser("Resource-Service", identity)
+	return svc, NewResourceController(svc, rest.db)
+}
+
+func (rest *TestResourceREST) SecuredControllerWithServiceAccount(serviceAccount account.Identity) (*goa.Service, *ResourceController) {
+	svc := testsupport.ServiceAsServiceAccountUser("Resource-Service", serviceAccount)
+	return svc, NewResourceController(svc, rest.Application)
+}
+
+func (rest *TestResourceREST) UnSecuredController() (*goa.Service, *ResourceController) {
+	svc := goa.New("Resource-Service")
+	return svc, NewResourceController(svc, rest.db)
+}
+
+/*
+ * This test will attempt to register a resource with an invalid account
+ */
+func (rest *TestResourceREST) TestFailRegisterResourceNonServiceAccount() {
+	testIdentity, err := testsupport.CreateTestIdentity(rest.DB, "TestRegisterResourceCreated-"+uuid.NewV4().String(), "TestRegisterResourceCreated")
+	require.Nil(rest.T(), err)
+
+	sa := account.Identity{
+		Username: "unknown-sa",
+	}
+	service, controller := rest.SecuredController(sa)
+
+	resourceDescription := "Resource description"
+	resourceID := ""
+	resourceScopes := []string{}
+
+	resourceOwnerID := testIdentity.ID
+
+	payload := &app.RegisterResourcePayload{
+		Description:      &resourceDescription,
+		Name:             "My new resource",
+		ParentResourceID: nil,
+		ResourceScopes:   resourceScopes,
+		ResourceID:       &resourceID,
+		ResourceOwnerID:  resourceOwnerID.String(),
+		Type:             "Area",
+	}
+
+	test.RegisterResourceUnauthorized(rest.T(), service.Context, service, controller, payload)
+}
+
+/*
+ * This test will attempt to register a resource with an invalid parent resource
+ */
+func (rest *TestResourceREST) TestFailRegisterResourceInvalidParentResource() {
+	testIdentity, err := testsupport.CreateTestIdentity(rest.DB, "TestRegisterResourceCreated-"+uuid.NewV4().String(), "TestRegisterResourceCreated")
+	require.Nil(rest.T(), err)
+
+	sa := account.Identity{
+		Username: "fabric8-wit",
+	}
+	service, controller := rest.SecuredControllerWithServiceAccount(sa)
+
+	resourceDescription := "Resource description"
+	resourceID := ""
+	resourceScopes := []string{}
+
+	resourceOwnerID := testIdentity.ID
+	parentResourceID := uuid.NewV4().String()
+
+	payload := &app.RegisterResourcePayload{
+		Description:      &resourceDescription,
+		Name:             "My new resource",
+		ParentResourceID: &parentResourceID,
+		ResourceScopes:   resourceScopes,
+		ResourceID:       &resourceID,
+		ResourceOwnerID:  resourceOwnerID.String(),
+		Type:             "Area",
+	}
+
+	test.RegisterResourceBadRequest(rest.T(), service.Context, service, controller, payload)
+}
+
+func (rest *TestResourceREST) TestRegisterResourceCreated() {
+
+	testIdentity, err := testsupport.CreateTestIdentity(rest.DB, "TestRegisterResourceCreated-"+uuid.NewV4().String(), "TestRegisterResourceCreated")
```

Comment:
You don't need to create an identity in the DB to use it as a service account, do you?
And it's better to create one regular identity and one service account in SetupSuite() method so they are available for all tests and you can reuse them. You could avoid copy-paste the same code in every test method then.